Add functions to interactive

minhtien-trinh opened the pull request:
As discussed with @LucaMarconato @melonora and @josenimo, I added two small functions to annotate the shapes/visualize the sample id. The two functions are:

1. `get_layer` grabs the layer name and returns the layer if it matches the user input. 
2. `add_text_to_polygons` adds annotations/sample names visually to the chosen polygons/shapes layer

When the user visualizes their data, annotations or simple sample id's could help in identifying samples and aid in analysis. 
Example usage would be: 

```
sample_id_list = gdf.index.tolist()
polygon_layer_name = 'Control'
interactive.add_text_to_polygons(polygon_layer_name, text_annotations)
```
1. Grab the sample id from the GeoDataFrame (index is used by default when linking a table with a shapes element)
2. Define the shapes layer (name of the spatial element shapes)
3. annotate (for now the default is: font size 10, red, center)

Ideally it will look like this:
![image](https://github.com/user-attachments/assets/defdb3e1-73c3-4165-9dbe-fe44b9db785b)

LucaMarconato commented:
Hi @minhtien-trinh thanks for your contribution! I reviewed the code and it looks great to me.

I ask you please two small adjustments before merging:
- the `codecov` job doesn't pass, could you please add some tests to cover the new code added?
- please update the changelog file. You can add the PR number `#315` and your GitHub handle `@minhtien-trinh`, so this information will be parsed by the release notes.

Thanks!

minhtien-trinh commented:
@LucaMarconato I wrote tests, modified the changelog and changed my code so that pre-commit succeeds. For some reason docs fails now and I'm uncertain why. Can you help me?

Czaki commented:
It looks like bug in run on CI. 
You may try to re-trigger CI pushing empty commit. 

```bash
git commit -m "trigger ci" --allow-empty
```

minhtien-trinh commented:
@Czaki I tried using the command you sent me but I the docs seem to fail still. Any idea what may be the issue? 

When opening the [link](https://readthedocs.org/projects/napari-spatialdata/builds/25863882/) for the docs, there is a message regarding an update on October 7th:
![image](https://github.com/user-attachments/assets/ec3088f6-736e-4812-8123-21bb2bf49dc2)

 Could [this](https://about.readthedocs.com/blog/2024/07/addons-by-default/) be the cause for the error?

minhtien-trinh commented:
@LucaMarconato Ready to merge :)

Czaki commented on src/napari_spatialdata/_interactive.py:
```diff
@@ -100,3 +101,28 @@ def run(self) -> None:
+
+    def get_layer(self, layer_name: str) -> Image | Labels | Points | Shapes | None:
+        """Get a layer by name."""
+        for layer in self._viewer.layers:
+            if layer.name == layer_name:
+                return layer
+        return None
```
```suggestion
        try:
            return self._viewer.layers[layer_name]
        except KeyError:
            return None
```

Czaki commented on src/napari_spatialdata/_interactive.py:
```diff
@@ -100,3 +101,28 @@ def run(self) -> None:
+
+    def add_text_to_polygons(self, layer_name: str, text_annotations: list[str]) -> None:
+        """Add text annotations to a polygon layer."""
+        polygon_layer = self.get_layer(layer_name)
+        if polygon_layer:
+            if len(text_annotations) != len(polygon_layer.data):
+                raise ValueError(
+                    f"The number of text annotations must match the number of polygons. "
+                    f"Polygons: {len(polygon_layer.data)}, Text: {len(text_annotations)}."
+                )
+            polygon_layer.text = {
+                "string": text_annotations,
+                "size": 10,
+                "color": "red",
+                "anchor": "center",
+            }
+        else:
+            raise ValueError(f"Polygon layer '{layer_name}' not found.")
```
Early return and reduction of indent increases readability 

```suggestion
        polygon_layer = self.get_layer(layer_name)
        if not polygon_layer:
            raise ValueError(f"Polygon layer '{layer_name}' not found.")
        if len(text_annotations) != len(polygon_layer.data):
            raise ValueError(
                f"The number of text annotations must match the number of polygons. "
                f"Polygons: {len(polygon_layer.data)}, Text: {len(text_annotations)}."
            )
        polygon_layer.text = {
            "string": text_annotations,
            "size": 10,
            "color": "red",
            "anchor": "center",
        }

            
```

Czaki reviewed (commented):
Two remarks for performance and readability.

LucaMarconato commented:
Thanks for the contribution!
